scripts/lind_run

```diff
@@ -31,13 +61,78 @@ if [[ -z "${REPO_ROOT}" || ! -d "${REPO_ROOT}" ]]; then
   exit 2
 fi
 
-WASMTIME_PROFILE="${WASMTIME_PROFILE:-release}"
-WASMTIME_BIN="${REPO_ROOT}/src/wasmtime/target/${WASMTIME_PROFILE}/wasmtime"
-# Fallback to release if selected profile isn't built yet
-if [[ ! -x "${WASMTIME_BIN}" ]]; then
-  ALT="${REPO_ROOT}/src/wasmtime/target/release/wasmtime"
-  [[ -x "${ALT}" ]] && WASMTIME_BIN="${ALT}"
+# --- Parse optional --wasmtime-profile flag (and strip it from args) ---
+# We only consume flags that are meaningful to this wrapper script.
+# All remaining arguments are forwarded verbatim to `wasmtime run`.
+WASMTIME_PROFILE=""
+PASSTHRU_ARGS=()
+
+while [[ $# -gt 0 ]]; do
+  case "$1" in
+    -h|--help)
+      print_help
+      exit 0
+      ;;
+    --wasmtime-profile=*)
+      WASMTIME_PROFILE="${1#*=}"
+      shift
+      ;;
+    *)
+      PASSTHRU_ARGS+=("$1")
+      shift
+      ;;
+  esac
+done
+
+# Validate profile value early to fail fast and clearly.
+case "${WASMTIME_PROFILE}" in
+  ""|release|debug) : ;;
+  *)
+    echo "error: invalid --wasmtime-profile '${WASMTIME_PROFILE}' (expected release or debug)" >&2
+    exit 2
+    ;;
+esac
+
+# --- Locate candidate wasmtime binaries ---
+TARGET_DIR="${REPO_ROOT}/src/wasmtime/target"
+REL_BIN="${TARGET_DIR}/release/wasmtime"
+DBG_BIN="${TARGET_DIR}/debug/wasmtime"
+
+# Track existence/executability explicitly to avoid ambiguous fallbacks.
+rel_ok=0; dbg_ok=0
+[[ -x "${REL_BIN}" ]] && rel_ok=1
+[[ -x "${DBG_BIN}" ]] && dbg_ok=1
+
+# --- Decide profile/binary ---
+# Policy:
+#   - If user explicitly specifies a profile, honor it strictly.
+#   - Otherwise:
+#       * Use the only available profile if exactly one exists.
```

Comment:
I think this isn't a great way to go about this since we can easily be confused that were using one while actually using the other. Id rather there be some sort of warning/error if there's a mismatch and we fix the environments properly.

Reply:
having the separate containers have the proper builds as defaults as well as not have dual version builds at once